### PR TITLE
Add more e2e tests service client retries

### PR DESCRIPTION
**Pull request by CIPop**
- Adding retries for IoT Hub Service Client method invoke in E2E tests.
- Remove unused `edison` CMake option.

### CODE REVIEW

**Review by danewalton — APPROVED** [review]
Good update thanks Cristian!

**Comment by CIPop** [comment]
Thanks @danewalton ! 

Got one DPS precondition failed and one DPS throttle condition. I'll increase the randomized back-off time interval to lower the likelihood of this happening.
If that's still not going to lower the failure rates I'll invest more time to create a retry mechanism for DPS tests as well.

```
Error: Time:Wed Feb 23 20:51:54 2022 File:/__w/1/s/provisioning_client/src/prov_device_ll_client.c Func:prov_transport_process_json_reply Line:677 Unsuccessful json encountered: {"errorCode":409203,"trackingId":"04f230de-eef2-44d5-969a-66f69a9faef6","message":"Precondition failed.","timestampUtc":"2022-02-23T20:51:54.87485Z"}
```

File: iothub_client/tests/common_device_method_e2e/iothubclient_common_device_method_e2e.c [hunk 1]
```diff
@@ -208,20 +208,68 @@ static int DeviceMethodWithUploadCallback(const char* method_name, const unsigne
...
+    size_t responsePayloadSize = 0;
+    IOTHUB_DEVICE_METHOD_RESULT invokeResult = IOTHUB_DEVICE_METHOD_ERROR;
 
     if (moduleId != NULL)
     {
-        LogInfo("IoTHubDeviceMethod_InvokeModule deviceId='%s', moduleId='%s'", deviceId, moduleId);
-        IOTHUB_DEVICE_METHOD_RESULT invokeResult = IoTHubDeviceMethod_InvokeModule(serviceClientDeviceMethodHandle, deviceId, moduleId, METHOD_NAME, payload, TIMEOUT, &responseStatus, &responsePayload, &responsePayloadSize);
+        for (int tryCounter = 0; tryCounter < TEST_METHOD_INVOKE_MAX_RETRIES; tryCounter++)
+        {
+            LogInfo("(Try %d) IoTHubDeviceMethod_InvokeModule deviceId='%s', moduleId='%s'", tryCounter + 1, deviceId, moduleId);
```
Comment (ericwol-msft) [inline iothub_client/tests/common_device_method_e2e/iothubclient_common_device_method_e2e.c:220]: too verbose?

Comment (CIPop) [reply iothub_client/tests/common_device_method_e2e/iothubclient_common_device_method_e2e.c:220]: I'll remove these and leave only the error reports in place.

File: iothub_client/tests/common_device_method_e2e/iothubclient_common_device_method_e2e.c [hunk 2]
```diff
@@ -208,20 +208,68 @@ static int DeviceMethodWithUploadCallback(const char* method_name, const unsigne
...
+
         ASSERT_ARE_EQUAL(IOTHUB_DEVICE_METHOD_RESULT, IOTHUB_DEVICE_METHOD_OK, invokeResult, "Service Client IoTHubDeviceMethod_InvokeModule failed");
     }
     else
     {
-        LogInfo("IoTHubDeviceMethod_Invoke deviceId='%s'", deviceId);
-        IOTHUB_DEVICE_METHOD_RESULT invokeResult = IoTHubDeviceMethod_Invoke(serviceClientDeviceMethodHandle, deviceId, METHOD_NAME, payload, TIMEOUT, &responseStatus, &responsePayload, &responsePayloadSize);
+        for (int tryCounter = 0; tryCounter < TEST_METHOD_INVOKE_MAX_RETRIES; tryCounter++)
+        {
+            LogInfo("(Try %d) IoTHubDeviceMethod_Invoke deviceId='%s'", tryCounter + 1, deviceId);
```
Comment (ericwol-msft) [inline iothub_client/tests/common_device_method_e2e/iothubclient_common_device_method_e2e.c]: too verbose

**Review by ericwol-msft — APPROVED** [review]

**Review by ericwol-msft — APPROVED** [review]

File: iothub_client/tests/common_device_method_e2e/iothubclient_common_device_method_e2e.c (hunk 2 above)
Comment (CIPop) [reply iothub_client/tests/common_device_method_e2e/iothubclient_common_device_method_e2e.c]: Removed.

**Pull request merged**